library/network_connections.py

```diff
@@ -1148,6 +1148,61 @@ def connection_create(self, connections, idx, connection_current=None):
                     s_ip4.add_route(rr)
                 else:
                     s_ip6.add_route(rr)
+            for r in ip["routing_rules"]:
+                routing_rule = NM.IPRoutingRule.new(r["family"])
+                if r["priority"] is not None:
+                    NM.IPRoutingRule.set_priority(routing_rule, r["priority"])
+                if r["action"]:
+                    for i, action in enumerate(
+                        [
+                            "to-table",
+                            "blackhole",
+                            "prohibit",
+                            "unreachable",
+                        ]
+                    ):
+                        if action == r["action"]:
+                            # check the link below for the enum value of supported action
+                            # https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/include/uapi/linux/fib_rules.h#n88
+                            NM.IPRoutingRule.set_action(routing_rule, (1, 6, 7, 8)[i])
+                if r["dport_start"]:
```

Comment:
it seems ugly in the YAML to have fields `*_start` and `*_end`.

Have just a `dport|sport|uid` field, which can be a single number (integer or as string). Additionally, it can be a string in the form of `r'^ *([0-9]+) *- *([0-9]+) *$'`.

Reply:
maybe it could also a two element list:
```
    - dport: 
      - 55
      - 66
```
but I think that
```
     - dport: "55-66"
```
is better API in the playbook.